infinite-tracing/src/main/java/com/newrelic/ConnectionHeaders.java

```diff
@@ -24,13 +19,11 @@ public void set(String newRunToken, Map<String, String> headers) {
         newHeaders.put("agent_run_token", newRunToken);
         newHeaders.put("license_key", licenseKey);
         this.headers = newHeaders;
-
-        logger.log(Level.INFO, "New Relic connection successful. Attempting connection to the Trace Observer.");
-        connectionStatus.reattemptConnection();
```

Comment:
This was a leaky abstraction. ConnectionHeaders should be a simple data carrier. Its methods should not be concerned with connection management.